ext/spl/spl_directory.c

```diff
@@ -2570,19 +2493,51 @@ PHP_METHOD(SplFileObject, getCsvControl)
 }
 /* }}} */
 
-/* {{{ Portable file locking */
-FileFunction(flock)
+static int flock_values[] = { LOCK_SH, LOCK_EX, LOCK_UN };
+
+/* {{{ Portable file locking, copy pasted from ext/standard/file.c flock() function.
+ * This is done to prevent this to fail if flock is disabled via disable_functions */
```

Comment:
Is that a bug or a feature? If someone disabled flock(), I imagine they also wouldn't want SplFileInfo::flock() to work, no?

Reply:
That's a good question, but most SPL methods are not disabled when the corresponding functional version is disabled. That's why I did this, moreover this was considered as an internal "bug" so if we want to revert this I think any function which is disabled should then disable it's SPL counterpart IMHO

Reply:
Either way, copy and paste doesn't look like a good idea to me.